src/java.base/share/classes/java/lang/Double.java

```diff
@@ -200,6 +200,139 @@
+ * exponent boundaries, the region is asymmetrical and larger on the
+ * side with the larger exponent.) For example, for {@code 0.1f}, the
+ * region can be computed as follows:
+ *
+ * <br>// Numeric values listed are exact values
+ * <br>oneTenthApproxAsFloat = 0.100000001490116119384765625;
+ * <br>ulpOfoneTenthApproxAsFloat = Math.ulp(1.0f) = 7.450580596923828125E-9;
+ * <br>// Numeric range that is converted to the float closest to 0.1, _excludes_ endpoints
+ * <br>(oneTenthApproxAsFloat - &frac12;ulpOfoneTenthApproxAsFloat, oneTenthApproxAsFloat + &frac12;ulpOfoneTenthApproxAsFloat) =
+ * <br>(0.0999999977648258209228515625, 0.1000000052154064178466796875)
+ *
+ * <p>In particular, a correctly rounded decimal to binary conversion
+ * of any string representing a number in this range, say by {@link
+ * Float#valueOf(float)}, will be converted to the same value:
+ *
+ * {@snippet lang="java" :
+ * Float.valueOf("0.0999999977648258209228515625000001"); // rounds up to oneTenthApproxAsFloat
+ * Float.valueOf("0.099999998");                          // rounds up to oneTenthApproxAsFloat
+ * Float.valueOf("0.1");                                  // rounds up to oneTenthApproxAsFloat
+ * Float.valueOf("0.100000001490116119384765625");        // exact conversion
+ * Float.valueOf("0.100000005215406417846679687");        // rounds down to oneTenthApproxAsFloat
+ * Float.valueOf("0.100000005215406417846679687499999");  // rounds down to oneTenthApproxAsFloat
+ * }
+ *
+ * <p>An analogous range can be constructed similarly for the {@code
+ * double} type based on the exact value of {@code double}
+ * approximation to {@code 0.1d} and the numerical value of {@code
+ * Math.ulp(0.1d)} and likewise for other particular numerical values
+ * in the {@code float} and {@code double} types.
+ *
+ * <p>As seen in the above conversions, compared to the exact result
+ * the same floating-point value as a result can be:
+ * <ul>
+ * <li>greater than the exact result
+ * <li>less than the exact result
+ * <li>equal to the exact result
+ * </ul>
+ *
+ * A floating-point value doesn't "know" if it was the result of
+ * rounding up, or rounding down, or an exact operation; it contains
+ * no history of how it was computed. Consequently, the sum of
+ * {@snippet lang="java" :
+ * 0.1f + 0.1f + 0.1f + 0.1f + 0.1f + 0.1f + 0.1f + 0.1f + 0.1f + 0.1f;
+ * // Numerical value of computed sum: 1.00000011920928955078125,
+ * // the next floating-point value larger than 1.0f, equal to Math.nextUp(1.0f).
+ * }
+ * or
+ * {@snippet lang="java" :
+ * 0.1d + 0.1d + 0.1d + 0.1d + 0.1d + 0.1d + 0.1d + 0.1d + 0.1d + 0.1d;
+ * // Numerical value of computed sum: 0.99999999999999988897769753748434595763683319091796875,
+ * // the next floating-point value smaller than 1.0d, equal to Math.nextDown(1.0d).
+ * }
+ *
+ * should <em>not</em> be expected to be exactly equal to 1.0, but
+ * only close to 1.0. Consequently the following code is an infinite loop:
```

Comment:
I think "Consequently" is a conjunctive adverb here, so should be followed by a comma. Or perhaps just drop it, since the previous sentence also starts with "Consequently, "